configure.ac

```diff
@@ -10,6 +10,9 @@ AC_PREREQ([2.69])
 AC_INIT([cglm], [0.6.1], [info@recp.me])
 AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects serial-tests])
 
+# Don't use the default cflags (-O2 -g), we set ours manually in Makefile.am.
+: ${CFLAGS=""}
```

Comment:
Is there any other default flags? We are removing them all, is there any side effect to remove them entirely?

Reply:
There's no other default flags. The Autoconf documentation recommends this: <https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/C-Compiler.html#index-AC_005fPROG_005fCC-861>

> If output variable `CFLAGS` was not already set, set it to `-g -O2` for the GNU C compiler (`-O2` on systems where GCC does not accept `-g`), or `-g` for other compilers. If your package does not like this default, then it is acceptable to insert the line `: ${CFLAGS=""}` after `AC_INIT` and before `AC_PROG_CC` to select an empty default instead.

Since we don't want `-g` or `-O2`, an empty default makes more sense. The user can still override this by setting the `CFLAGS` variable explicitly.